src/Mouf/Database/MagicQuery.php

```diff
@@ -304,15 +305,18 @@ private function getSchemaAnalyzer()
                 throw new MagicQueryMissingConnectionException('In order to use MagicJoin, you need to configure a DBAL connection.');
             }
 
-            $this->schemaAnalyzer = new SchemaAnalyzer($this->connection->getSchemaManager(), $this->cache, $this->getConnectionUniqueId($this->connection));
+            $this->schemaAnalyzer = new SchemaAnalyzer($this->connection->createSchemaManager(), $this->cache, $this->getConnectionUniqueId($this->connection));
         }
 
         return $this->schemaAnalyzer;
     }
 
     private function getConnectionUniqueId(Connection $connection): string
     {
-        return hash('md4', $connection->getHost().'-'.$connection->getPort().'-'.$connection->getDatabase().'-'.$connection->getDriver()->getName());
+        $connectionParams = $connection->getParams();
+        \assert(\array_key_exists('host', $connectionParams));
+        \assert(\array_key_exists('port', $connectionParams));
```

Comment:
Do we want to use `assert` here? Or should we have explicit error in case thoses are not set?

Reply:
This is only to satisfy static analysis, these will be set once already